pipeline_steps/release.groovy

```diff
@@ -0,0 +1,24 @@
+void tag(String version, String ref){
+  print "Tagging ${version} at ${ref}"
+  sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+    sh """/bin/bash -xe
+      git fetch --all --tags
+      git tag -a -m \"${version}\" \"${version}\" origin/${ref}
+      git push --tags origin
+    """
+  }
+}
+
+void update_rc_branch(String repo, String mainline, String rc){
+  print "Resetting ${rc} to head of ${mainline}"
+  sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+    sh """/bin/bash -xe
+      git fetch --all --tags
+      git checkout -b ${rc} || git checkout ${rc}
+      git reset --hard origin/${mainline}
+      git push -f origin ${rc}:${rc}
```

Comment:
This seems like it's going to conflict with GitHub protected branches, or is the intention to stop using that feature?

Reply:
Good point, it will conflict if the rc branches are protected, which they probably should be.

Reply:
I've made the rc branch protected. Is there any way of allowing the jenkins user to push a tag with protection on? Alternatively, could the protection be removed by the jenkins user, the tag pushed, then protection added back?

Reply:
I've checked rpc-openstack and rpc-maas, and we don't include admins in branch restrictions, so as long as Jenkins is an admin, force push should work. Will do some testing.

Reply:
The only reason admins aren't included is because someone forgot to add that flag, the always were previously.

Reply:
Turning off branch protection does loose all settings (which checks are required, who can push etc). So it may be better to toggle include-admins before and after pushing.